platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -465,15 +465,34 @@ + (instancetype)mgl_expressionWithJSONObject:(id)object {
             return [NSExpression expressionForVariable:argumentObjects.firstObject];
         } else if ([op isEqualToString:@"case"]) {
             NSPredicate *conditional = [NSPredicate mgl_predicateWithJSONObject:argumentObjects.firstObject];
-            NSExpression *trueExpression = [NSExpression mgl_expressionWithJSONObject:argumentObjects[1]];
+            NSExpression *trueExpression;
+            NSInteger rightBranchIndex = 2;
+
+            if ([argumentObjects[1] isKindOfClass:[NSArray class]] ) {
+                NSInteger length = 1;
+                for (NSInteger index = 1; index < argumentObjects.count - 2; index++) {
+                    length++;
+                    if (![argumentObjects[index] isKindOfClass:[NSArray class]] && ![argumentObjects[index + 1] isKindOfClass:[NSArray class]]) {
+                        break;
+                    }
+                }
+                NSArray *trueObjects = [@[@"case"] arrayByAddingObjectsFromArray:
+                                         [argumentObjects subarrayWithRange:NSMakeRange(1, length)]];
+                trueExpression = [NSExpression mgl_expressionWithJSONObject:trueObjects];
+                rightBranchIndex = length + 1;
+            } else {
+                trueExpression = [NSExpression mgl_expressionWithJSONObject:argumentObjects[1]];
+            }
+            
             NSExpression *falseExpression;
-            if (argumentObjects.count > 3) {
+            if ([argumentObjects[rightBranchIndex] isKindOfClass:[NSArray class]] ) {
```

Comment:
For #11007, we need to implement a new custom NSExpression function that can represent multiple branches. `TERNARY()` only supports “if…then”, and “else”, but we need “else if…then” as well.

If NSExpession were to support this feature natively, I would expect it to have initializers similar to `+[NSDictionary dictionaryWithObjects:forKeys:]` and `+[NSDictionary dictionaryWithObjectsAndKeys:]`, in other words:

* `+[NSExpression expressionForTrueExpressions:forConditionals:falseExpression:]`
* `+[NSExpression expressionForTrueExpressionsAndConditionals:falseExpression:]`

We could implement these methods in this PR to help address #11016 and remove the single-conditional initializer in https://github.com/mapbox/mapbox-gl-native/pull/11278#discussion_r174855147.

Once #11015 is implemented, I think the underlying implementation could become clearer, something like `MGL_TERNARY(firstConditional, firstExpression, secondConditional, secondExpression, …, falseExpression)`. In the meantime, the existing `FUNCTION()` syntax should suffice.